addon/services/key-manager.js

```diff
@@ -79,88 +82,97 @@ export default Ember.Service.extend({
     });
   },
 
-  handler(event) {
-    if (get(this, 'isDestroyed') || get(this, 'isDestroying')) {
-      return;
-    }
-    const {
-      data,
-      keyCode,
-    } = event;
+  findMatchingCombo(event) {
+    try {
+      const { data, keyCode } = event;
+      const { eventName = null } = data;
+
+      modifierKeys.forEach((key) => {
+        const prop = `${key}Key`;
+        set(this, prop, event[prop]);
+      });
 
-    if (!modifierKeyCodes.includes(keyCode)) {
-      get(this, 'downKeys').addObject(keyCode);
+      if (!modifierKeyCodes.includes(keyCode)) {
+        get(this, 'downKeys').addObject(keyCode);
+      }
+
+      return this._findComboByName(eventName);
+    } finally {
+      this._clearExecutionKeys(event);
     }
+  },
 
-    modifierKeys.forEach((key) => {
-      const prop = `${key}Key`;
-      set(this, prop, event[prop]);
-    });
+  handler(event) {
+    if (get(this, 'isDestroyed') || get(this, 'isDestroying')) { return; }
+
+    const combo = this.findMatchingCombo(event);
+    const runLoopGuard = !get(this, 'matchFound');
 
-    if (data) {
-      const { eventName } = data;
-      const combo = this._findComboByName(eventName);
-      const runLoopGuard = !get(this, 'matchFound');
-
-      if (combo && runLoopGuard) {
-        set(this, 'matchFound', true);
-        run.next(() => {
-          set(this, 'matchFound', false);
-        });
-
-        const isNotOnInput = inputElements.every(e => !$(document.activeElement).is(e));
-        if (!get(combo, 'disableOnInput') || isNotOnInput) {
-          const callback = get(combo, 'callback');
-          if (callback) {
-            callback(event);
-          }
+    if (combo && runLoopGuard) {
+      set(this, 'matchFound', true);
+      run.next(() => {
+        set(this, 'matchFound', false);
+      });
+
+      const isNotOnInput = inputElements.every(e => !$(document.activeElement).is(e));
+      if (!get(combo, 'disableOnInput') || isNotOnInput) {
+        const callback = get(combo, 'callback');
+        if (callback) {
+          callback(event);
         }
       }
     }
+  },
 
+  executionKeys: Ember.computed(function() {
```

Comment:
two nits - we use the object syntax for computed properties and we destructure one level deep so `computed` instead of `Ember.computed`